### PR TITLE
scala-cli: init at 0.0.7

### PR DESCRIPTION
**Pull request by kubukoz**
This originates from https://github.com/kubukoz/scala-cli.nix, wanted to make it more discoverable :)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - aarch64-linux
  - [x] x86_64-darwin
  - aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] ~Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))~ not applicable
- [ ] ~Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`~ - not applicable
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] ~(Package updates) Added a release notes entry if the change is major or breaking~ N/A
  - [ ] ~(Module updates) Added a release notes entry if the change is significant~ N/A
  - [ ] ~(Module addition) Added a release notes entry if adding a new NixOS module~ N/A
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/development/tools/build-managers/scala-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        sha256 = asset.sha256;
+      };
+
+  unpackPhase = ''
+    cat $src | gzip -d > scala-cli
+  '';
+
+  nativeBuildInputs = [
+    zlib
+    stdenv.cc.cc
```
[the remark this answers is not in the document]
Comment (kubukoz) [reply pkgs/development/tools/build-managers/scala-cli/default.nix:37]: oh. I'll update that, then

**Comment by kubukoz** [comment]
@supersandro2000 @AndersonTorres is this okay to merge?

File: pkgs/development/tools/build-managers/scala-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    x86_64-linux = {
+      asset = "scala-cli-x86_64-pc-linux.gz";
+      sha256 = "1xdkvjfw550lpjw5fsrv7mbnx5i8ix8lrxcd31yipm8p9g4vjcdn";
+    };
+  };
+in
+stdenv.mkDerivation {
+  pname = "scala-cli";
+  inherit version;
+  buildInputs = [ coreutils autoPatchelfHook installShellFiles zlib stdenv.cc.cc ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/scala-cli/default.nix:19]:
```suggestion
  nativeBuildInputs = [ autoPatchelfHook installShellFiles ];

  buildInputs = [ coreutils zlib stdenv.cc.cc ];
```

**Comment by kubukoz** [comment]
Is that all?

**Comment by AndersonTorres** [comment]
@kubukoz what makeSetupHook is doing in that code?

**Comment by kubukoz** [comment]
> @kubukoz what makeSetupHook is doing in that code?

Wasting disk space, apparently 😅 it's not used anymore, removed.

**Comment by AndersonTorres** [comment]
LGTM, LGTBorg.

**Comment by kubukoz** [comment]
Thanks!

**Comment by SuperSandro2000** [comment]
New packages are usually not backported.

**Comment by kubukoz** [comment]
alright :)

**Pull request merged**